pkg/pubsub/dapr/dapr.go

```diff
@@ -0,0 +1,83 @@
+package dapr
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/dapr/go-sdk/client"
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/connection"
+)
+
+type ClientConfig struct {
+	// Name of the component to be used for pub sub messaging
+	Component string `json:"component"`
+}
+
+// Dapr represents driver for interacting with pub sub using dapr.
+type Dapr struct {
+	// Array of clients to talk to different endpoints
+	client client.Client
+
+	// Name of the pubsub component
+	pubSubComponent string
+}
+
+const (
+	Name = "dapr"
+)
+
+func (r *Dapr) Publish(ctx context.Context, data interface{}, topic string) error {
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		return fmt.Errorf("error marshaling data: %w", err)
+	}
+
+	err = r.client.PublishEvent(context.Background(), r.pubSubComponent, topic, jsonData)
+	if err != nil {
+		return fmt.Errorf("error publishing message to dapr: %w", err)
+	}
+
+	return nil
+}
+
+func (r *Dapr) CloseConnection() error {
+	return nil
+}
+
+func (r *Dapr) UpdateConnection(ctx context.Context, config interface{}) error {
```

Comment:
can we add unit tests for this? can be a follow-up

Reply:
agree as follow up.